src/ucs/datastruct/string_set.c

```diff
@@ -109,16 +109,14 @@ ucs_status_t ucs_string_set_print_sorted(const ucs_string_set_t *sset,
                                          const char *sep)
 {
     const char **sorted_strings;
-    ucs_status_t status;
     size_t idx, count;
     char *str;
 
     /* allocate a temporary array to hold the sorted strings */
     count          = kh_size(sset);
     sorted_strings = ucs_calloc(count, sizeof(*sorted_strings), "string_set");
     if (sorted_strings == NULL) {
-        status = UCS_ERR_NO_MEMORY;
-        goto out;
+        return UCS_ERR_NO_MEMORY;
```

Comment:
why removed, i see the 'out' label was used?

Reply:
just once, without any action except return. Typically we do return in such case thru the code

Reply:
IMO a better practice is to have single return point in this case, even if no cleanup action is required

Reply:
why? Then we also have to initialize status in the case of success. Was not our rule return immediately when the fault is found?

Reply:
if we already use goto for error handling in a particular function, it 's cleaner all error handling would go through same place. in case we want to add more init steps, or trace where the function fails

Reply:
it contradicts with several of our rules:
- unneeded var initializations ( status)
- early return on fail

This label made sense before, when there was another one, but really looks like rudiment now.  
I really do not think it is worth reverting it back in this particular small function, but ok - will revert.

Reply:
ah, right, i missed that out_free_array was removed
so let's keep it as is